tests/unit_test/linux/README.md

```diff
@@ -0,0 +1,142 @@
+# Unit Testing with CMOCK setting up guide
+
+## Introduction to Unit Testing with AFR
+Unit Testing with AFR uses the **cmake build** framework.<br>
+Code coverage is achieved with **gcov/lcov** and results are displayed in: *build/coverage/*<br>
+The executables are located in *build/bin* <br>
+
+The build is tested on Linux and Windows with WSL. <br>
+
+
+## Introduction to CMOCK
+**CMOCK** is a mocking  framework, that automatically generates mocks.
+It gives the ability to set expectations and implement stubs/callbacks.
+This will allow you to call the functions to be tested and the **CMOCK** generated code will do all the rest for you.<br>
+
+For example suppose your module accesses the internet using sockets, but you
+want to test just your module (unit) without testing/calling any socket API
+function as this could be unpredictable, slow, hard to simulate all possible
+error scenarios and return codes.<br>
+In order to achieve that you tell **CMOCK** to mock the header file where the
+socket's APIs reside. For simplicity we will assume the only function used is
+*func*, **CMOCK** will generate the following mocks for you:
+
+```
+func_Ignore[AndReturn]([return_val]);
+func_ExpectAnyArgs[AndReturn]([return_val]);
+func_Expect[AndReturn](param1, param2,..., paramN ,[return_val])
+func_StubWithCallback(function_callback)
+func_ExpectWithArray[AndReturn](param1, [depth1], param2, [depth2], ..., paramN, [depthN], [return_val])
+func_ReturnThruPtr_[parameter name](ret_pointer)
+func_ReturnArrayThruPtr_parameeter(parameter, len)
+func_ReturnMemThruPtr_[parameter name](size, ret_memory)
+func_IgnoreArg_[parameter name]()
+```
+For a detailed explanation about these mocks check the official documentation:<br>
+The **CMOCK** official website click [here] (http://www.throwtheswitch.org/cmock)<br>
+The **CMOCK** git hub repository click [here] (https://github.com/ThrowTheSwitch/CMock)<br>
+
+## How to build and execute
+To build first go to the AFR root directory
+```
+$ cd afr_root
+$ cmake -B build -DAFR_ENABLE_UNIT_TESTS=on -DBOARD=linux -DVENDOR=pc -DCOMPILER=linux-gcc -DCMAKE_BUILD_TYPE=Debug
+$ cd build
+$ make
+$ make coverage
+```
+## Setting up a new Unit Testing module in AFR
+To Setup a module for Unit Testing, as an example we will follow a walkthrough
+approach for **lwip_secure_sockets** which are located in *libraries/abstractions/secure_sockets*. <br>
+
+1. Open the file *tests/unit_test/linux/CMakeLists.txt* for editing.<br>
+    Make sure at around line 15 the the Unit Test directory location exists in this case **../../../libraries** if not add it<br>
+
+``` cmake
+   add_subdirectory(../../../libraries libraries)
```

Comment:
Should this just be `add_subdirectory(../../../libraries ~~libraries~~)`

Reply:
i am not sure if both of them work, the current one is ../../../libraries libraries
it is working just fine